lib/ttfunk/table/os2.rb

```diff
@@ -143,6 +143,15 @@ class OS2 < Table
 
       UNICODE_MAX = 0xFFFF
       UNICODE_RANGES = UNICODE_BLOCKS.keys.freeze
+      LOWERCASE_START = 'a'.ord
+      LOWERCASE_END = 'z'.ord
+      LOWERCASE_COUNT = (LOWERCASE_END - LOWERCASE_START) + 1
+      CODEPOINT_SPACE = 32
+      WEIGHT_SPACE = 166
+      WEIGHT_LOWERCASE = [
+        64, 14, 27, 35, 100, 20, 14, 42, 63, 3, 6, 35, 20,
+        56, 56, 17, 4, 49, 56, 71, 31, 10, 18, 3, 18, 2
```

Comment:
I got these from Microsoft's Font-Validator project: https://github.com/HinTak/Font-Validator/blob/bf3a4be60d8616338444aeb4872f222f239c043c/OTFontFileVal/val_OS2.cs#L142

Reply:
Here's how it described in [the spec](https://docs.microsoft.com/en-us/typography/opentype/spec/os2#xavgcharwidth):

> The value for xAvgCharWidth _(ed: this field)_ is calculated by obtaining the arithmetic average of the width of all non-zero width glyphs in the font. Furthermore, it is strongly recommended that implementers do not rely on this value for computing layout for lines of text, especially for cases where complex scripts are used.

Furthermore:

> Versions 0 to 2: When first defined, the specification was biased toward Basic Latin characters, and it was thought that the xAvgCharWidth value could be used to estimate the average length of lines of text. A formula for calculating xAvgCharWidth was provided using frequency-of-use weighting factors for lowercase letters a – z. This method of calculating the value of this field was superseded in OpenType 1.4 with the introduction of version 3 of the OS/2 table.

The way I read it, is that the values correspond to frequencies in English text ([Wikipedia](https://en.wikipedia.org/wiki/Letter_frequency) seem to support this hypothesis) but it's not required to follow it.

* I'm fine leaving it in but please add a comment as to what those numbers are.
* Do we follow the spec in the versioning department?

Reply:
Yeah, that all makes sense. I was coding TTFunk's implementation against Font-Validator and trying to get a "green" report for subsetted fonts, which is why I ported over the average char width stuff (the report includes a few failures otherwise). Will definitely add a comment and excerpts from the spec.

Yes, we do follow the spec for versioning, although we _don't_ support os/2 v3.

Reply:
OK. Please add a comment with some context for these numbers.

Reply:
@camertron Please add this comment and we're good to go.

Reply:
@pointlessone I added a comment a little while ago, were you thinking of adding other specific info?

Reply:
My bad. Must've been looking at an old tab or something.